mkosi/gentoo.py

```diff
@@ -0,0 +1,482 @@
+        custom_profile_path = root / CUSTOM_PROFILE_PATH
+        custom_profile_path.mkdir(exist_ok=True)
+        custom_profile_path.joinpath("use.force").write_text(
+            dedent(
+                """\
+                    -split-bin
+                    -split-sbin
+                    -split-usr
+                    """
+            )
+        )
+
+        package_use = self.portage_cfg_dir / "package.use"
+        package_use.mkdir(exist_ok=True)
+
+        self.baselayout_use = package_use.joinpath("baselayout")
+        self.baselayout_use.write_text("sys-apps/baselayout build\n")
+        package_use.joinpath("grub").write_text("sys-boot/grub device-mapper truetype\n")
+        package_use.joinpath("systemd").write_text(
+            # repart for usronly
+            dedent(
+                """\
+                # sys-apps/systemd http
+                # sys-apps/systemd cgroup-hybrid
+
+                # MKOSI: Failed to open "/usr/lib/systemd/boot/efi": No such file or directory
+                sys-apps/systemd gnuefi
+
+                # sys-apps/systemd -pkcs11
+                # sys-apps/systemd importd lzma
+                sys-apps/systemd homed cryptsetup -pkcs11
+                # MKOSI: usronly
+                sys-apps/systemd repart
+                # sys-apps/systemd -cgroup-hybrid
+                # sys-apps/systemd vanilla
+                # sys-apps/systemd policykit
+                # MKOSI: make sure we're init (no openrc)
+                sys-apps/systemd sysv-utils
+                # sys-fs/lvm2 device-mapper-only -thin
+                """
+            )
+        )
+
+    def mkosi_conf(self, root: Path) -> None:
+        package_env = self.portage_cfg_dir / "package.env"
+        ebuild_sh_env_dir = root / EBUILD_SH_ENV_DIR
+        package_env.mkdir(exist_ok=True)
+        ebuild_sh_env_dir.mkdir(exist_ok=True)
+
+        # apply whatever we put in mkosi_conf to runs invokation of emerge
+        package_env.joinpath("mkosi.conf").write_text("*/*    mkosi.conf\n")
+
+        # we use this so we don't need to touch upstream files
+        # we also use this for documenting build environment as much as possible
+
+        emerge_vars_str = ""
+        for k, v in self.emerge_vars.items():
+            emerge_vars_str += f'{k}="$' + "{" + k + "}" + f' {v}"\n'
+
+        ebuild_sh_env_dir.joinpath("mkosi.conf").write_text(
+            dedent(
+                f"""\
+                    # MKOSI: these were used during image creation...
+                    # and some more! see under package.*/
+                    #
+                    # usrmerge (see all under profile/)
+                    {emerge_vars_str}
+                    """
+            )
+        )
+
+    def invoke_emerge(
+        self,
+        args: CommandLineArguments,
+        root: Path,
+        inside_stage3: bool = True,
+        pkgs: List[str] = list(),
+        actions: List[str] = list(),
+        opts: List[str] = list(),
```

Comment:
Uh, no, no, no. Mutable default arguments are bad, even if it doesn't matter in this case. Please use `Sequence[str] = ()`.

Reply:
i need `__add__` for passing them to `emerge_main`. doing it goofy way for now:
`emerge_main(list(pkgs) + list(opts) + list(actions) + PREFIX_OPTS + self.emerge_default_opts)`

any suggestions?

Reply:
`emerge_main([*pkgs, *opts, *actions, *PREFIX_OPTS, *emerge_default_opts])` ?